Realm/Realm/Sync/Session.cs

```diff
@@ -185,5 +225,34 @@ internal void CloseHandle(bool waitForShutdown = false)
         }
 
         internal void ReportErrorForTesting(int errorCode, string errorMessage, bool isFatal) => Handle.ReportErrorForTesting(errorCode, errorMessage, isFatal);
+
+        internal static void RaiseError(Session session, Exception error)
+        {
+            var args = new ErrorEventArgs(error);
+            Error?.Invoke(session, args);
+        }
+
+        private void SubscribeNotifications()
+        {
+            var cb = new Action<string>((propertyName) => NotifyPropertyChanged(propertyName));
+            var callbackHandle = GCHandle.Alloc(cb, GCHandleType.Weak);
+            var nativeTokens = Handle.RegisterPropertyChangedCallbacks(GCHandle.ToIntPtr(callbackHandle));
```

Comment:
Again, let's allocate a weak handle to the managed session rather than the callback.

Reply:
I thought that the managed session was a recommendation mostly because there was not in place a generic mechanism like this. So it'd have helped with other fields to raise notifications for. Since this does not seem to be the case, could you explain the advantages of passing the whole `SessionHandle` rather than an action. Does it have to do with lifecycle?